app/src/main/java/gateway/controller/CtrlFileList.java

```diff
@@ -0,0 +1,100 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.soap.request.ReqFileList;
+import gateway.soap.response.File;
+import gateway.soap.response.ResFileList;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlFileList
+{
+	public static ResFileList file_list (ReqFileList args)
+	{
+		// Create a new instance of ResFileList to store the results
+		ResFileList resFileList = new ResFileList ();
+		UUID userUUID;
+		String url;
+
+		// User authentication
+		ResStatus resAuth = ServiceAuth.authenticate (args.token);
+		if (resAuth.error) {
+			return ResStatus.downCast (ResFileList.class, resAuth);
+		}
+
+		// Get the user's UUID from the authentication token
+		userUUID = UUID.fromString (ServiceAuth.tokenGetClaim (args.token, "uuid"));
+
+		// Build the URL for the request.
+		if (args.location == null || args.location.equals (new UUID (0L, 0L))) {
```

Comment:
The second check is not needed, if it happens to be an invalid UUID the metadata will let's us know through an error